game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/CasualtyOrderOfLosses.java

```diff
@@ -276,13 +297,13 @@ static void clearOolCache() {
       }
       amphibHashCode *= 31;
       key =
-          player.getName()
+          parameters.player.getName()
               + "|"
-              + battlesite.getName()
+              + parameters.battlesite.getName()
               + "|"
-              + defending
+              + parameters.defending
               + "|"
-              + amphibious
+              + parameters.amphibious
               + "|"
               + targetsHashCode
               + "|"
```

Comment:
I'm seeing this code snippet an odd amount of times.
Thoughts about extracting this to a helper method?

Reply:
There are more areas of copy / paste than this too, code climate does a pretty decent job pointing them out.

This update is just preliminary, intentionally not modifying too much of the existing code yet. It needs to be done iteratively. I'm hoping as well that this hashing function can be removed as part of one those iterations.